File: C++/M01_ProgrammingBasics/L04_ForLoop/Exercises/Solutions/P03_OddEvenPosition.cpp
#include <iostream>
#include <climits>
#include <cmath>
using namespace std;

int main() {

	int n;
	cin >> n;

	double oddSum = 0;
	//double oddMin = 1000000000.0;
	double oddMin = DBL_MAX;
	//double oddMax = -1000000000.0;
	double oddMax = DBL_MIN;
	double evenSum = 0;
	//double evenMin = 1000000000.0;
	double evenMin = DBL_MAX;
	//double evenMax = -1000000000.0;
	double evenMax = ;
	double evenMax = ;

	for (int i = 1; i <= n; i++) {
		double input;
		cin >> input;

		if (i % 2 == 0) {
			evenSum += input;
			evenMin = fmin(evenMin, input);
			evenMax = fmax(evenMax, input);
		}
		else {
			oddSum += input;
			oddMin = fmin(oddMin, input);
			oddMax = fmax(oddMax, input);
		}
	}

	cout.setf(ios::fixed);
	cout.precision(2);

	cout << "OddSum=" << oddSum << "," << endl;
	if (oddSum == 0) {
		cout << "OddMin=No," << endl;
		cout << "OddMax=No," << endl;
	}
	else {
		cout << "OddMin=" << oddMin << "," << endl;
		cout << "OddMax=" << oddMax << "," << endl;
	}
	cout << "EvenSum=" << evenSum << "," << endl;
	if (evenSum == 0) {
		cout << "EvenMin=No," << endl;
		cout << "EvenMax=No" << endl;
	}
	else {
		cout << "EvenMin=" << evenMin << "," << endl;
		cout << "EvenMax=" << evenMax << endl;
	}

	return 0;
}

File: Python/M01_ProgrammingBasics/L04_ForLoop/Exercises/Solutions/P03_OddEvenPosition.py
import sys
number = int(input())
max_odd = -sys.maxsize
max_even = -sys.maxsize
min_odd = sys.maxsize
min_even = sys.maxsize

odd = 0
even = 0
odd_sum = 0
even_sum = 0

for each in range(number):
    current = float(input())

    if each % 2 == 1:
        even_sum += current

        if current > max_even:
            max_even = current
        if current < min_even:
            min_even = current
    else:
        odd_sum += current

        if current > max_odd:
            max_odd = current
        if current < min_odd:
            min_odd = current

print(f"OddSum={odd_sum:.2f},")
if min_odd == sys.maxsize:
    print('OddMin=No,')
else:
    print(f"OddMin={min_odd:.2f},")
if max_odd == -sys.maxsize:
    print('OddMax=No,')
else:
    print(f"OddMax={max_odd:.2f},")

print(f"EvenSum={even_sum:.2f},")
if min_even == sys.maxsize:
    print('EvenMin=No,')
else:
    print(f"EvenMin={min_even:.2f},")
if max_even == -sys.maxsize:
    print('EvenMax=No')
else:
    print(f"EvenMax={max_even:.2f}")

